syllabus.md

```diff
@@ -0,0 +1,32 @@
+# Syllabus
+
+The following topics relate to code quality.
+
+- [ ] SW Interfaces
+- [x] The build pattern
+- [ ] Anonymous function (lambda)
+  - Closure
+- [ ] The SOLID principles
+  - [ ] Single responsibility principle
+  - [ ] Open-Closed principle
+  - [ ] Liskov Substitution principle
+  - [ ] Interface segregation principle
+  - [ ] Dependency inversion principle
+- [ ] Immutable functions
+- [ ] Naming and comments
+- [ ] Code Structure
+- [ ] Testing
+  - [ ] Test doubles
+  - [ ] Tips for testable code part 1
+  - [ ] Tips for testable code part 2
+  - [ ] Unit, Integration or Functional?
+  - [ ] Test Driven Development
+- [ ] Source control
```

Comment:
Please consider adding a lecture on composing PRs, and strategies for moving forward "controversial" PRs.

Reply:
Good idea, I will add it.

Reply:
Done